src/ducks/modules/mock.js

```diff
@@ -7,23 +7,27 @@ import { nodeAttributesProperty } from './network';
 
 const MOCK_GENERATE_NODES = 'MOCK/GENERATE_NODES';
 
-const generateNodes = (howMany = 0) =>
+const keyForVarName = (vars, name) => {
+  const entry = Object.entries(vars).find(([, val]) => val.name === name);
+  return entry && entry[0];
+};
+
+const generateNodes = (variableDefs, typeKey, howMany = 0) =>
   (dispatch) => {
     times(howMany, () => {
       const firstName = faker.name.firstName();
       const lastName = faker.name.lastName();
       const age = faker.random.number({ min: 16, max: 99 });
 
       return dispatch(sessionsActions.addNodes({
-        type: 'person',
+        type: typeKey,
         promptId: 'mock',
         stageId: 'mock',
         [nodeAttributesProperty]: {
-          name: `${firstName} ${lastName}`,
-          nickname: lastName,
-          age,
+          [keyForVarName(variableDefs, 'name')]: `${firstName} ${lastName}`,
```

Comment:
I think this should have a similar interface to `getNodeAttribtuesWithNamesResolved` (sp.), just in reverse?

```
return dispatch(sessionsActions.addNodes({
  type: typeKey,
  promptId: 'mock',
  stageId: 'mock'
  [nodeAttributesProperty]: getAttributesWithIds(
    variableDefinitions,
    {
      name: `${firstName} ${lastName}`,
      nickname: lastName,
      age,
    }
  ),
})
```

Or maybe it could be at the node level?:

```
  getNodeWithIdAttributes(variableDefinitions, nodeProperties, nodeAttributes);
```

Reply:
This is now `getNodeWithIdAttributes` in the network module. With the latest changes to mock generation, it's never called. However, I'm leaving it in since it'll be useful when importing external data down the line.